packages/trpc/server/template-router/router.ts

```diff
@@ -0,0 +1,116 @@
+import { TRPCError } from '@trpc/server';
+
+import { createDocumentFromTemplate } from '@documenso/lib/server-only/template/create-document-from-template';
+import { createTemplate } from '@documenso/lib/server-only/template/create-template';
+import { createTemplateFromDocument } from '@documenso/lib/server-only/template/create-template-from-document';
+import { deleteTemplate } from '@documenso/lib/server-only/template/delete-template';
+import { duplicateTemplate } from '@documenso/lib/server-only/template/duplicate-template';
+
+import { authenticatedProcedure, router } from '../trpc';
+import {
+  ZCreateDocumentFromTemplateMutationSchema,
+  ZCreateTemplateFromDocumentMutationSchema,
+  ZCreateTemplateMutationSchema,
+  ZDeleteTemplateMutationSchema,
+  ZDuplicateTemplateMutationSchema,
+} from './schema';
+
+export const templateRouter = router({
+  createTemplate: authenticatedProcedure
+    .input(ZCreateTemplateMutationSchema)
+    .mutation(async ({ input, ctx }) => {
+      try {
+        const { title, templateDocumentDataId } = input;
+
+        return await createTemplate({
+          title,
+          userId: ctx.user.id,
+          templateDocumentDataId,
+        });
+      } catch (err) {
+        console.error(err);
+
+        throw new TRPCError({
+          code: 'BAD_REQUEST',
+          message: 'We were unable to create this template. Please try again later.',
+        });
+      }
+    }),
+
+  createDocumentFromTemplate: authenticatedProcedure
+    .input(ZCreateDocumentFromTemplateMutationSchema)
+    .mutation(async ({ input, ctx }) => {
+      try {
+        const { templateId } = input;
+
+        return await createDocumentFromTemplate({
+          templateId,
+          userId: ctx.user.id,
+        });
+      } catch (err) {
+        throw new TRPCError({
+          code: 'BAD_REQUEST',
+          message: 'We were unable to create this document. Please try again later.',
+        });
+      }
+    }),
+
+  createTemplateFromDocument: authenticatedProcedure
```

Comment:
I think we don't need the two way binding right now with how can create templates. Lets just stick with `template => document` and not allow for `document => template`

Reply:
Got it. I'll remove this

Reply:
Just to save you from some testing headaches 💚